File: java/Easy/_867_TransposeMatrix.java
package Easy;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

import org.testng.annotations.Test;

public class _867_TransposeMatrix {
    public int[][] transpose(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        int[][] res = new int[n][m];
        for(int i = 0; i < m; i++)
            for(int j = 0; j < n; j++)
                res[j][i] = matrix[i][j];
        return res;
    }

    @Test
    public void test(){
        assertArrayEquals(new int[][]{{1,4,7},{2,5,8},{3,6,9}}, transpose(new int[][]{{1,2,3},{4,5,6},{7,8,9}}));
    }
}

File: Python3/_867_TransposeMatrix.py


from typing import List

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        result = [[0]*len(matrix) for _ in range(len(matrix[0]))]
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                result[j][i] = matrix[i][j]
        return result  

test = Solution()
test.transpose([[2,4,-1],[-10,5,11],[18,-7,6]])
